src/core/string_set.cc

```diff
@@ -88,12 +92,20 @@ unsigned StringSet::AddBatch(absl::Span<std::string_view> span, uint32_t ttl_sec
     Prefetch(hash[i]);
   }
 
+  // update ttl if legacy_saddex_keepttl is off (which is default). This variable is intended for
+  // SADDEX, but this method is called from SADD as well, where ttl is set to UINT32_MAX value,
+  // which results in has_ttl being false. This means that ObjUpdateExpireTime is never called from
+  // SADD code path even when update_ttl is true.
+  const bool update_ttl = !absl::GetFlag(FLAGS_legacy_saddex_keepttl);
```

Comment:
absl::GetFlag uses atomic, so it is better to use cache the value of flags specially for operations we want to optimise performance, in command flow f.e

Reply:
Do you mean something like below in the method scope

```
const thread_local bool update_ttl = !absl::GetFlag(FLAGS_legacy_saddex_keepttl);
```

This does make testing legacy behavior tricky as I cannot reset the value during test but I guess this is fairly simple logic so might be fine to keep out of test? 

Is there an example somewhere in the code base where a flag is cached so I can use it for reference?

Reply:
What you wrote is good for this case.
We usually store it in object member but there is no object here which you can store.
I agree we can not change the value while testing, in this case I suggest to test only the default behaviour

Reply:
SGTM. Changed the variable storage type and the test.